lib/rspec/core/example_group.rb

```diff
@@ -408,16 +408,6 @@ def self.subclass(parent, description, args, registration_collection, &example_g
 
       # @private
       def self.set_it_up(description, args, registration_collection, &example_group_block)
-        # Ruby 1.9 has a bug that can lead to infinite recursion and a
-        # SystemStackError if you include a module in a superclass after
-        # including it in a subclass: https://gist.github.com/845896
-        # To prevent this, we must include any modules in
-        # RSpec::Core::ExampleGroup before users create example groups and have
-        # a chance to include the same module in a subclass of
-        # RSpec::Core::ExampleGroup. So we need to configure example groups
-        # here.
-        ensure_example_groups_are_configured
```

Comment:
@JonRowe 
With this cleanup, I broke the following example in `rspec-expectations`:
```ruby
RSpec.describe do
  RSpec::Matchers.define(:be_awesome) { 1 }
  # define(:be_awesome) { 1 }

  context "when accessed via an alias that is not included in failure messages" do
    alias_method :be_fantastic, :be_awesome
  end
end
```

with:
```
Failure/Error: alias_method :be_fantastic, :be_awesome

NameError:
  undefined method `be_awesome' for class `#<Class:0x00007ff6540f4dd8>'
# ./spec/rspec/matchers/define_negated_matcher_spec.rb:6:in `alias_method'
```

If you define the matcher with plain `define`, or `matcher(:be_awesome) { ... }` which are available in the example group scope, example passes:
```ruby
RSpec.describe do
  define(:be_awesome) { 1 }

  context "when accessed via an alias that is not included in failure messages" do
    alias_method :be_fantastic, :be_awesome
  end
end
```

It does fail outside of examples only, and only on `alias_method`. The matcher defined with `RSpec::Matchers.define` is available:
```ruby
RSpec.describe do
  RSpec::Matchers.define :be_awesome do
    match(&:awesome)
  end

  it do
    expect(double(awesome: true)).to be_awesome
  end
end
```

Root-level `RSpec::Matchers.define` works fine:
```ruby
RSpec::Matchers.define(:be_one) do
  match { |actual| actual == 1 }
end

RSpec.describe do
  it { expect(1).to be_one }
end
```

I couldn't find any mention that `RSpec::Matchers.define` can be used inside of the example group in docs: https://relishapp.com/rspec/rspec-expectations/v/3-9/docs/custom-matchers/define-a-custom-matcher.

What are your thoughts on this?
I'm not asking "why would anyone", pretty sure that someone used that if it was possible.
More like are you ok with removing such a possibility with a major version update?

Reply:
https://github.com/rspec/rspec-expectations/pull/1231/commits/83ed8fc6c83b25e4ae35fbbdbeccbc3cfd8d5a9f

Reply:
To be fair, there's a problem with `rspec-mocks`'s specs where aliases fail (both `alias` and `alias_method`):
```
        it_behaves_like "a named verifying double", "InstanceDouble" do
          alias :create_double :instance_double
          alias_method :create_double, :instance_double
```
I'll take a closer look.

Reply:
This is actually broken as well:
```ruby
RSpec::Matchers.define(:be_one) do
  match { |actual| actual == 1 }
end

RSpec.describe do
  alias_method :b, :be_one
  it { expect(1).to be_one }
end
```

I'll shuffle things around to make it all work.

Reply:
I'll leave this out for a follow-up.